### ISSUE TITLE
Setting LoaderOptimization reg key prevents IIS from starting

### ISSUE THREAD

**Issue by rlaveycal** [DataDog/dd-trace-dotnet#667]
**Describe the bug**

With APM >= 1.12 you can no longer set the `LoaderOptimization` registry key to work around #475/#592 because you get error
```
Failed to initialize the AppDomain:/LM/W3SVC/1/ROOT

Exception: System.InvalidOperationException

Message: The configuration system has already been initialized.

StackTrace:    at System.Configuration.ConfigurationManager.SetConfigurationSystem(IInternalConfigSystem configSystem, Boolean initComplete)
   at System.Web.Configuration.HttpConfigurationSystem.EnsureInit(IConfigMapPath configMapPath, Boolean listenToFileChanges, Boolean initComplete)
   at System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException)
   at System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel)
   at System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel)
   at System.Web.Hosting.ApplicationManager.CreateAppDomainWithHostingEnvironment(String appId, IApplicationHost appHost, HostingEnvironmentParameters hostingParameters)
   at System.Web.Hosting.ApplicationManager.CreateAppDomainWithHostingEnvironmentAndReportErrors(String appId, IApplicationHost appHost, HostingEnvironmentParameters hostingParameters)
```

**To Reproduce**
1. Install IIS with ASP.Net
1. Install APM/DD Agent
1. Set `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\.NETFramework\LoaderOptimization` to `1`
1. Access http://localhost

You'll get a 500 server error. Check app event log for details

**Expected behavior**
The web app starts

**Runtime environment (please complete the following information):**
- Instrumentation mode: MSI
- Tracer version: 1.13.2
- OS: Windows 2016
- CLR: Default (4.6.2)

**Additional context**

As an exercise I manually added `Datadog.Trace.AspNet` to the GAC and installed the module into IIS the normal way. Disabling the profiler allowed me to get ASP.Net traces. Attempts to have profiling enabled but the aspnet integration disabled (by adding `DD_DISABLED_INTEGRATIONS=AspNet` to the WAS environment reg key) failed.

**Comment by lucaspimentel** [comment]
Thanks for reporting this, @rlaveycal. This seems related to the changes made in 1.12.0 to enable the ASP.NET integration automatically (instead of requiring a NuGet package or changes to `web.config`).

Our next release (you can try out [the preview now](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.14.1-prerelease)) includes changes to prevent the original issue (#592) by automatically disabling an integration if we detect that instrumenting that assembly will cause the `SECURITY_E_INCOMPATIBLE_SHARE` exception to be thrown.

We won't consider that issue fixed until we can safely enable integrations without throwing the exception above and without the need for the `LoaderOptimization` workaround.

**Comment by rlaveycal** [comment]
I've just tried 1.14.2 on my test server and there's no change in behaviour for this issue.

**Comment by zacharycmontoya** [comment]
Hi @rlaveycal, thank you for your detailed report. Using that information I was able to track this down to a change in [v1.12.0](https://github.com/DataDog/dd-trace-dotnet/compare/v1.11.0..v1.12.0#diff-c3200d8114ed5d17efa5e815e658c893R47) that affects the initialization of our automatic instrumentation and I completed #713 to resolve this issue, which has further details if you're interested in taking a look. This issue will be fixed in the next release. Thanks!
